src/main/java/com/scaler/lld/design/assignments/prototype/Configuration.java

```diff
@@ -1,6 +1,14 @@
 package com.scaler.lld.design.assignments.prototype;
 
-public class Configuration {
+import lombok.NoArgsConstructor;
+import lombok.Setter;
+
+import java.util.HashMap;
+import java.util.Map;
+
+@NoArgsConstructor
+@Setter
+public class Configuration implements ConfigurationPrototypeRegistry, ClonableObject<Configuration> {
```

Comment:
The implementation here is correct, however it tends to violate SRP since the model and the registry is coupled.
Move the registry to aa separate class.